### PR TITLE
Add support for Variable and Node rules in `add_platform_rule.py`

### PR DESCRIPTION
**Pull request by yuumasato**
#### Description:

- Add support for platform rules with rule parametrization
- Add support for creation of node rules based on templates

#### Rationale:

- Supports the [Compliance-Operator Workshop](https://github.com/ComplianceAsCode/compliance-operator/pull/403)
- Makes it easier for new comers to create rules

### CODE REVIEW

**Comment by yuumasato** [comment]
The script should likely be renamed.

File: utils/add_platform_rule.py [hunk 1]
```diff
@@ -180,8 +211,32 @@ def which(program):
...
+            print("ERROR: The specified group '%s' doesn't exist in the '%s' directory" % (
+                args.group, OCP_RULE_DIR))
+            return 0
+
+    rule_path = os.path.join(group_path, args.rule)
+    rule_yaml_path = os.path.join(rule_path, 'rule.yml')
+
+    mkdir_p(rule_path)
+    with open(rule_yaml_path, 'w') as f:
+        var_list = [ f"    {k.strip()}" for k in args.template_vars.split(",")]
```
Comment (rhmdnd) [inline utils/add_platform_rule.py:227]: An alternative to formatting this way would be to put everything in a dictionary, then render the YAML using a library (like ruamel, or pyyaml).

**Review by rhmdnd — APPROVED** [review]
/lgtm

File: utils/add_platform_rule.py (hunk 1 above)
Comment (yuumasato) [reply utils/add_platform_rule.py:227]: I agree that the current approach is not ideal, I merely followed the existing one.

I can update the script to use a more yaml friendly way.

**Review by rhmdnd — APPROVED** [review]
/lgtm

**Pull request merged**